src/Application.cpp

```diff
@@ -18,6 +18,32 @@
 Application::Application(int& argc, char** argv) :
     QApplication(argc, argv)
 {
+    // Check 'DB4S_CONFIG_FILE environment variable exists
+    const char* env = getenv("DB4S_CONFIG_FILE");
+
+    // If 'DB4S_CONFIG_FILE' environment variable exists
+    if(env)
+    {
+        Settings::userConfiguration = true;
+        Settings::userConfigurationDir = QString::fromStdString(env);
+    }
+
+    for(int i=1;i<arguments().size();i++)
+    {
+        if(arguments().at(i) == "-c" || arguments().at(i) == "--config")
+        {
+            if(++i >= arguments().size())
+            {
+                qWarning() << qPrintable(tr("The -c/--config option requires an argument"));
+            } else {
+                if (env)
+                    qWarning() << qPrintable(tr("The user configuration file location is replaced with the argument value instead of the environment variable value."));
```

Comment:
Consider giving the variable name so user knows what variable to unset to stop the warning.